src/application/controllers/load-guardian-name.ts

```diff
@@ -0,0 +1,26 @@
+import { type GetGuardianName } from '@/domain/use-cases'
+import { success, type HttpRequest, type HttpResponse, serverError } from '../helpers'
+import { type Controller } from '../protocols'
+
+export class LoadGuardianNameController implements Controller {
+  private readonly getGuardianName: GetGuardianName
+
+  constructor ({ getGuardianName }: LoadGuardianNameController.Dependencies) {
+    this.getGuardianName = getGuardianName
+  }
+
+  async handle (httpRequest: HttpRequest): Promise<HttpResponse> {
+    try {
+      const guardianName = await this.getGuardianName.load(httpRequest.userId as string)
```

Comment:
Is there any possibility of this function return `undefined`? I think not because the only possibility is if auth token has wrong id inside of (so login route is broken in this case, or security issues), is it necessary handle this failure case?

Reply:
I put the possible return "undefined" because this method will bring me a guardian of the database. As the database method can return "undefined" I thought it good to put this possibility